wallet/datastore.go

```diff
@@ -559,20 +559,32 @@ func (pg *Postgres) IncreaseLinkingLimit(ctx context.Context, providerLinkingID
 
 // LinkWallet links a wallet together
 func (pg *Postgres) LinkWallet(ctx context.Context, ID string, userDepositDestination string, providerLinkingID uuid.UUID, anonymousAddress *uuid.UUID, depositProvider string) error {
-	sublogger := logger(ctx).With().
-		Str("wallet_id", ID).
-		Logger()
+	sublogger := logger(ctx).With().Str("wallet_id", ID).Logger()
 
 	// create tx
 	tx, err := createTx(ctx, pg)
 	if err != nil || tx == nil {
-		sublogger.Error().Err(err).
-			Msg("error creating tx for wallet linking")
+		sublogger.Error().Err(err).Msg("error creating tx for wallet linking")
 		return fmt.Errorf("failed to create tx for wallet linking: %w", err)
 	}
+
+	err = waitAndLock(ctx, tx, providerLinkingID)
```

Comment:
if you move to the model shown below, you will want to put this code below https://github.com/brave-intl/bat-go/pull/1033/files#diff-561e26c4278cfea65e512b2590a23c624ebdaf56f0dedb6b9e92ef45f83a4e45R577-R578
https://github.com/brave-intl/bat-go/blob/master/wallet/datastore.go#L384-L390

Reply:
I'd suggest instrumenting this in prometheus, so we can watch for an explosion in count of advisory locks in pg.